### PR TITLE
Share master yum repos with agents

### PR DESCRIPTION
**Pull request by joshsamuelson**
This will work on 32 and 64bit agents once we have the yum repos for both platforms cached on the master.

### CODE REVIEW

File: manifests/master/yum_server.pp [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class classroom::master::yum_server {
+  file { $classroom::repo_base_path:
+    ensure => directory,
+  }
+  $classroom::repos.each |$repo_name,$repo_path| {
```
Comment (carthik) [inline manifests/master/yum_server.pp]: where is classroom::repos defined?

File: manifests/agent/yum_repos.pp [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class classroom::agent::yum_repos {
+  $classroom::yum_repos.each |$repo, $value| {
```
Comment (carthik) [inline manifests/agent/yum_repos.pp]: classroom::yum_repos - where is it defined and was this meant to be classroom_repos as in manifests/master/yum_server.pp?

Comment (joshsamuelson) [inline manifests/agent/yum_repos.pp]: oops this is a typo.  I fixed it on my testing machine, just a sec. I'll push it up.

File: manifests/master/yum_server.pp (hunk 1 above)
Comment (joshsamuelson) [inline manifests/master/yum_server.pp]: It's in params.pp so it gets inherited by init.pp.

Comment (carthik) [inline manifests/master/yum_server.pp]: I can see $repositories in params.pp and no $repos in init.pp or params.pp.

File: manifests/params.pp [hunk 3]
```diff
@@ -108,4 +108,12 @@
       fail("Your VM is out of date. ${download}/puppet-student.ova/\n\n")
     }
   }
+
+  $repo_base_path = '/opt/puppetlabs/server/data/packages/public/yum'
+  $repos = {
```
Comment (joshsamuelson) [inline manifests/params.pp]: Here is repos

**Comment by carthik** [comment]
Looks good - Shall I merge this or are we waiting till yum repos for both architectures are cached on the master?

**Comment by joshsamuelson** [comment]
@carthik I think you can go ahead an merge it.  It won't break existing functionality.

**Pull request merged**